### PR TITLE
Add validation notification create

### PR DESCRIPTION
**Pull request by mithuna**
## Description
 This PR adds input validation functionality to create notification.

## Steps to Test
Make a POST request to localhost:3000/api/notifications and provide different data types for input variables.

### CODE REVIEW

**Review by cecilia-donnelly — COMMENTED** [review]
I'll wait to give an official review until after the other PR is merged, so that this is one is more clearly defined, but the validation definitely works!  Thanks for adding this.

File: src/validators/validateCreateNotificationParams.ts
Comment (jasonaowen) [inline src/validators/validateCreateNotificationParams.ts:8]: I turns out that, as written, Joi will allow floating point numbers, which will then cause PostgreSQL to throw an error:

```
$ curl -v -X POST -H "Content-Type: application/json" --data '{"toUserId":4.3,"notificationType":"not an integer"}' http://localhost:3000/api/notifications | jq .
(trimmed)
< HTTP/1.1 500 Internal Server Error
(trimmed)
{
  "error": {
    "name": "TinyPgError",
    "queryContext": {
      "id": "c8926922-8000-4c4e-9f7c-2e329e63af93",
      "sql": "INSERT \nINTO notifications (to_user_id, notification_type) \nVALUES ($1, $2)\nRETURNING notification_id\n",
      "start": 1615243221034,
      "name": "insert-notification",
      "params": {
        "to_user_id": 4.3,
        "notification_type": "not an integer"
      },
      "submit": 1615243221039,
      "wait_duration": 5,
      "active_duration": 30,
      "end": 1615243221069,
      "duration": 35,
      "error": {
        "length": 104,
        "name": "error",
        "severity": "ERROR",
        "code": "22P02",
        "file": "numutils.c",
        "line": "323",
        "routine": "pg_strtoint32"
      },
      "data": null
    }
  }
}
```

It looks like Joi has an [integer requirement](https://joi.dev/api/?v=17.4.0#numberinteger), which is pretty straightforward to add:

```suggestion
  toUserId: Joi.number().integer().min(1).required(),
```

**Review by jasonaowen — CHANGES_REQUESTED** [review]
Great job, @mithuna! This looks really good!

I was able to put on my QA hat and find a value that Joi allowed but PostgreSQL didn't, but it took some doing; everything else I could think of was correctly blocked by Joi, with a super helpful error message!

Could you also please squash the commits together? I think there's only one logical change here; the history doesn't benefit from having a separate linter commit or from having a integers-only commit.

**Comment by jasonaowen** [comment]
Oh, and to make the link explicit: this PR builds on #12, and so the diff is showing the commit from #12 in addition to the commits specific to this branch. This should not be merged until after #12 is merged. (speaking of validations... :joy:)

File: src/validators/validateCreateNotificationParams.ts
Comment (mithuna) [reply src/validators/validateCreateNotificationParams.ts:8]: Oh thank you for testing that. Yes will make changes.

**Comment by jasonaowen** [comment]
It looks like GitHub hasn't noticed that some of the commits in this branch are now on `main` since you merged #12. @mithuna, would you mind rebasing this branch onto `main`?

**Review by jasonaowen — APPROVED** [review]
Looks good! Great work, @mithuna!

**Pull request merged**
